### PR TITLE
Add text-to-image and text generation providers

### PR DESCRIPTION
**Pull request by julien-nc**
And while I was on it, add psalm and php-cs checks and fix all errors.

### CODE REVIEW

File: lib/TextToImage/TextToImageProvider.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+	/**
+	 * @inheritDoc
+	 */
+	public function generate(string $prompt, array $resources): void {
+		$nbOutputs = count($resources);
+		$waitingForAlready = 0;
+		$maxWaitTime = 60 * 60;
+		try {
+			$prediction = $this->replicateAPIService->createImagePrediction($prompt, $this->userId, $nbOutputs, '768x768');
```
Comment (bigcat88) [inline lib/TextToImage/TextToImageProvider.php:46]: Can we make this customizable?

Most modern models outputs not only square 768x768 and 1024x1024, but 1536x864,  2048x1152, 1024x576 for wide screens..

Comment (marcelklehr) [reply lib/TextToImage/TextToImageProvider.php:46]: Not immediately in this PR, long term, sure

File: lib/TextProcessing/FreePromptProvider.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+			throw new \RuntimeException('Replicate\'s text generation failed with: ' . $e->getMessage());
+		}
+	}
+
+	public function getTaskType(): string {
+		return FreePromptTaskType::class;
+	}
+
+	public function getExpectedRuntime(): int {
+		return 60 * 60 * 24;
```
Comment (marcelklehr) [inline lib/TextProcessing/FreePromptProvider.php:70]: :raised_eyebrow:

Comment (marcelklehr) [reply lib/TextProcessing/FreePromptProvider.php:70]: Should be 60*60, looking at https://github.com/nextcloud/integration_replicate/pull/13/files#diff-b2195ef468317f8efe4759c18a81b0323b5ea9557c81b6b8e2687648f5382c3dR34 ?

**Review by marcelklehr — APPROVED** [review]

**Review by bigcat88 — APPROVED** [review]

File: lib/TextToImage/TextToImageProvider.php (hunk 1 above)
Comment (julien-nc) [reply lib/TextToImage/TextToImageProvider.php]: It's kind of needed to add parameters. As parameters are different for each models and we can't predict them. It's now possible to set any extra parameter. Up to the NC admins to read model-specific docs on Replicate's website. These extra params are global and apply to all users for tasks using the Replicate image/text generation provider.

It's also possible to choose a model by name or version code. Up to the admins to explore Replicate's website to find models.

Comment (marcelklehr) [reply lib/TextToImage/TextToImageProvider.php]: Yeah, sorry about the ambiguous answer

**Pull request merged**